### PR TITLE
Part 2 of jobs mapper implementation

### PR DESCRIPTION
**Pull request by sdatko**
- Implement jobs override mechanism
- Consider jobs properties in templater
- Update config map with initial values

### CODE REVIEW

File: znoyder/config.yml [hunk 1]
```diff
@@ -71,34 +71,135 @@ exclude:
...
+            AODH_TEST_DRIVERS: 'mysql'
+
+  'python-ovsdbapp':
+    'osp-17.0':
+      'openstack-tox-py39':
+        voting: false
+        vars:
+          rhos_release_args: '17.0'
+
+  'python-swiftclient':
```
Comment (cescgina) [inline znoyder/config.yml:182]: This job doesn't seem to have any additional variable? Should it be in the add section?

File: znoyder/mapper.py [hunk 2]
```diff
@@ -109,7 +109,21 @@ def add_jobs(jobs, project, tag) -> list:
...
+        for tag_specifier in override_map[project_specifier]:
+            if not match(tag, tag_specifier):
+                continue
+
+            override_map_jobs = override_map[project_specifier][tag_specifier]
+            for job_entry in override_map_jobs.items():
+                for new_job in job_from_map_entry(job_entry):
+                    for index, job in enumerate(jobs):
+                        if match(job.job_name, new_job.job_name):
+                            jobs[index].job_data = new_job.job_data
```
Comment (cescgina) [inline znoyder/mapper.py:125]: This overrides completely the job variables, which makes sense. I wonder however, if there might be any case when there are variables from upstream that we want to keep.

**Review by cescgina — APPROVED** [review]
Overall looks great, I left a couple of question for clarification

File: znoyder/config.yml (hunk 1 above)
Comment (sdatko) [reply znoyder/config.yml:182]: No, this one exists in upstream, however with `voting` set to `true`. This overrides it to `false` :-)

File: znoyder/mapper.py (hunk 2 above)
Comment (sdatko) [reply znoyder/mapper.py:125]: I will merge the existing change as is, to have override map in place, and the improvement will be implemented as separate change.

**Pull request merged**